packages/cli/src/lib/Prompt.ts

```diff
@@ -31,6 +31,12 @@ export type ValidateFunction<T> = (input: string) => Result<T, Error>;
  * A an interface that promps for different types of user input.
  */
 export interface Prompt {
+  /**
+   * Prints a message to the user.
+   * @param message the message to be printed.
+   */
+  printMessage(message: string): Promise<void>;
```

Comment:
I find it strange that you don't mark the interface method `async` (I know this is a TypeScript thing, not a mistake in your code). I guess the `Promise` does somewhat imply it, but the idea of having some subclass having an `async` implementation and some not seems weird to me.

Reply:
+1. I often forget and mark interfaces as `async` and get the compiler complaining about it.